gdal/ogr/ogrsf_frmts/ngw/ogrngwdriver.cpp

```diff
@@ -0,0 +1,294 @@
+        osUserPwdOption += osUserPwd;
+        papszOptions = CSLAddString(papszOptions, osUserPwdOption.c_str());
+    }
+    return papszOptions;
+}
+
+/*
+ * OGRNGWDriverDelete()
+ */
+static CPLErr OGRNGWDriverDelete( const char *pszName )
+{
+    NGWAPI::Uri stUri = NGWAPI::ParseUri(pszName);
+    CPLErrorReset();
+    if( !stUri.osNewResourceName.empty() )
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Unsupported url %s", pszName);
+        return CE_Failure;
+    }
+
+    if( stUri.osPrefix != "NGW" )
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Unsupported name %s", pszName);
+        return CE_Failure;
+    }
+
+    if( stUri.osResourceId == "0" )
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Cannot delete resource 0");
+        return CE_Failure;
+    }
+
+    char **papszOptions = GetHeaders();
+    // NGWAPI::Permissions stPermissions = NGWAPI::CheckPermissions(stUri.osAddress,
+    //     stUri.osResourceId, papszOptions, true);
+    // if( stPermissions.bResourceCanDelete )
+    // {
+        return NGWAPI::DeleteResource(stUri.osAddress, stUri.osResourceId,
+            papszOptions) ? CE_None : CE_Failure;
+    // }
+    // CPLError(CE_Failure, CPLE_AppDefined, "Operation not permitted.");
+    // return CE_Failure;
+}
+
+/*
+ * OGRNGWDriverRename()
+ */
+static CPLErr OGRNGWDriverRename( const char *pszNewName, const char *pszOldName )
+{
+    NGWAPI::Uri stUri = NGWAPI::ParseUri(pszOldName);
+    CPLErrorReset();
+    if( stUri.osPrefix != "NGW" )
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Unsupported name %s", pszOldName);
+        return CE_Failure;
+    }
+    CPLDebug("NGW", "Parse uri result. URL: %s, ID: %s, New name: %s",
+        stUri.osAddress.c_str(), stUri.osResourceId.c_str(), pszNewName);
+    char **papszOptions = GetHeaders();
+    // NGWAPI::Permissions stPermissions = NGWAPI::CheckPermissions(stUri.osAddress,
+    //     stUri.osResourceId, papszOptions, true);
+    // if( stPermissions.bResourceCanUpdate )
+    // {
+        return NGWAPI::RenameResource(stUri.osAddress, stUri.osResourceId,
+            pszNewName, papszOptions) ? CE_None : CE_Failure;
+    // }
+    // CPLError(CE_Failure, CPLE_AppDefined, "Operation not permitted.");
+    // return CE_Failure;
+}
+
+/*
+ * RegisterOGRNGW()
+ */
+
+void RegisterOGRNGW()
+{
+    if( GDALGetDriverByName( "NGW" ) != nullptr )
+    {
+        return;
+    }
+
+    GDALDriver *poDriver = new GDALDriver();
+
+    poDriver->SetDescription( "NGW" );
+    poDriver->SetMetadataItem( GDAL_DMD_LONGNAME, "NextGIS Web" );
+    poDriver->SetMetadataItem( GDAL_DCAP_RASTER, "YES" );
+    poDriver->SetMetadataItem( GDAL_DCAP_VECTOR, "YES" );
+    poDriver->SetMetadataItem( GDAL_DMD_SUBDATASETS, "YES" );
+    poDriver->SetMetadataItem( GDAL_DMD_HELPTOPIC, "drv_ngw.html" );
+    poDriver->SetMetadataItem( GDAL_DMD_CONNECTION_PREFIX, "NGW:" );
+    poDriver->SetMetadataItem( GDAL_DMD_CREATIONDATATYPES, "Byte" );
```

Comment:
there's no raster creation capability, so no need for CREATIONDATATYPES

Reply:
Fix per 4579aeab927d70ddab6142340bf5550c6789b42e